### ISSUE TITLE
CategoricalDomain contains method has unexpected behavior

### ISSUE THREAD

**Issue by k8si** [factorie/factorie#379]
tl;dr: I noticed that calling `contains` on `CategoricalDomain` has what I consider to be unexpected behavior and I'm not sure why this occurs or how to fix it.

In my opinion, `domain.index(value) >= 0`, `domain.categories.contains(value)` and `domain.contains(value)` should all return the same result, but they don't, as illustrated in the following slightly-modified version of `TestCategoricalDomain`:

    val domain = new CategoricalDomain[String](List("yes", "no"))
    domain.freeze()
    assert(domain.size == 2) // passes
    // I added the following lines
    assert(domain.index("yes") >= 0) //passes
    assert(domain.categories.contains("yes")) //passes
    assert(domain.contains("yes")) //fails

It seems like it's standard "FACTORIE insider knowledge" to just use one of `domain.index(value) >= 0` or `domain.categories.contains(value)` as a proxy for `contains`, but I think it'd be nice to either fix this issue or at least make it more obvious to the user that `contains` does not work so they should use a different method instead (e.g. throw an informative UnsupportedOperationException).

My first thought was to submit a PR for CategoricalDomain with something like:

    override def contains(category: C): Boolean = indexOrNegativeOne(category) >= 0

If I don't override, the compiler finds a name clash because of type erasure; if I do override, the compiler says the method overrides nothing. So the above is out, unless someone knows how to Scala their way around it.

I'm still not even sure why `contains` doesn't work in the first place, and I've been staring at the debugger for too long. So... I would welcome additional input/advice on how to deal with this small but thorny issue.
